exchanges/binance/binance.go

```diff
@@ -940,3 +941,138 @@ func (b *Binance) MaintainWsAuthStreamKey() error {
 		HTTPRecording: b.HTTPRecording,
 	})
 }
+
+// SetExchangeTolerances sets exchange tolerances for all assets
+func (b *Binance) SetExchangeTolerances() error {
+	var limits []exchange.MinMaxLevel
+
+	spot, err := b.GetExchangeInfo()
+	if err != nil {
+		return err
+	}
+
+	for x := range spot.Symbols {
+		var cp currency.Pair
+		cp, err = currency.NewPairFromStrings(spot.Symbols[x].BaseAsset,
+			spot.Symbols[x].QuoteAsset)
+		if err != nil {
+			return err
+		}
+		var assets []asset.Item
+		for y := range spot.Symbols[x].Permissions {
+			switch spot.Symbols[x].Permissions[y] {
+			case "SPOT":
+				assets = append(assets, asset.Spot)
+			case "MARGIN":
+				assets = append(assets, asset.Margin)
+			case "LEVERAGED": // leveraged tokens not available for spot trading
+			default:
+				return fmt.Errorf("unhandled asset type for tolerance loading %s",
+					spot.Symbols[x].Permissions[y])
+			}
+		}
+
+		for z := range assets {
+			if len(spot.Symbols[x].Filters) < 8 {
+				continue
+			}
+
+			limits = append(limits, exchange.MinMaxLevel{
+				Pair:             cp,
+				Asset:            assets[z],
+				MinPrice:         spot.Symbols[x].Filters[0].MinPrice,
+				MaxPrice:         spot.Symbols[x].Filters[0].MaxPrice,
+				StepPrice:        spot.Symbols[x].Filters[0].TickSize,
+				MultiplierUp:     spot.Symbols[x].Filters[1].MultiplierUp,
+				MultiplierDown:   spot.Symbols[x].Filters[1].MultiplierDown,
+				AveragePriceMins: spot.Symbols[x].Filters[1].AvgPriceMins,
+				MaxAmount:        spot.Symbols[x].Filters[2].MaxQty,
+				MinAmount:        spot.Symbols[x].Filters[2].MinQty,
+				StepAmount:       spot.Symbols[x].Filters[2].StepSize,
+				MinNotional:      spot.Symbols[x].Filters[3].MinNotional,
+				MaxIcebergParts:  spot.Symbols[x].Filters[4].Limit,
+				MarketMinimumQty: spot.Symbols[x].Filters[5].MinQty,
```

Comment:
Really a lame request but since everything else is min and max you could change this to MarketMinQty